packages/core/src/services/documents/updateListOfIntegrations.ts

```diff
@@ -0,0 +1,99 @@
+import Transaction from '../../lib/Transaction'
+import { IntegrationsRepository } from '../../repositories'
+import { documentIntegrationReferences } from '../../schema'
+import { and, eq } from 'drizzle-orm'
+import { getDocumentMetadata } from './scan'
+import { DocumentVersion } from '../../browser'
+import { LatitudePromptConfig } from '@latitude-data/constants/latitudePromptSchema'
+import { Result } from '../../lib/Result'
+
+function getToolIds(tools: LatitudePromptConfig['tools']): string[] {
+  if (!tools) return []
+
+  if (typeof tools === 'string') {
+    return [tools]
+  }
+
+  if (Array.isArray(tools)) {
+    return tools
+      .map((tool) => {
+        if (typeof tool === 'string') return [tool]
+
+        return Object.keys(tool)
+      })
+      .flat()
+  }
+
+  return Object.keys(tools)
+}
+
+function getIntegrationNames(toolsIds: string[]): string[] {
+  return toolsIds
+    .map((toolId) => {
+      const [integrationName, toolName] = toolId.split('/')
+      if (!integrationName?.length) return undefined // Return undefined if name is empty
+      if (!toolName?.length) return undefined // Return undefined if there is no '/' (not an integration tool)
+      return integrationName
+    })
+    .filter((name): name is string => name !== undefined) // Remove instances where there is no '/'
+    .filter((name, index, array) => array.indexOf(name) === index) // Remove duplicates
+}
+
+export async function updateListOfIntegrations(
+  {
+    workspaceId,
```

Comment:
should receive a workspace instance rather than workspaceId since you are using it for tenancy check in line 77